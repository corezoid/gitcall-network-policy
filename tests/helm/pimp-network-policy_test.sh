#!/usr/bin/env bash
$CZ helm template | sed -n '/pimp-network-policy\.yaml/,/Source\:/p'
