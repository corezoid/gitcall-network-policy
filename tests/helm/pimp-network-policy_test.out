# Source: gitcall_network_policy_release/charts/gitcall_network_policy/templates/pimp-network-policy.yaml
apiVersion: crd.projectcalico.org/v1
kind: NetworkPolicy
metadata:
  name: RELEASE-NAME-gitcall-pimp-network-policy
  namespace: "aNamespace"
  labels:
    release: "RELEASE-NAME"
    app: "gitcall"
    tier: "network_policy"
spec:
  selector: tier == 'pimp'
  types:
    - Egress
    - Ingress
  ingress:
    # Gitcall
    - action: Allow
      protocol: TCP
      source:
        selector: tier == 'gitcall'
      destination:
        ports:
          - 8080
    # Prometheus
    - action: Allow
      protocol: TCP
      source:
        namespaceSelector: projectcalico.org/name == 'monitoring'
        selector: app == 'prometheus'
      destination:
        ports:
          - 9100
  egress:
    # DNS
    - action: Allow
      protocol: UDP
      destination:
        namespaceSelector: projectcalico.org/name == 'kube-system'
        selector: k8s-app == 'kube-dns'
        ports:
          - 53
    # Kube API
    # rule for apiserver with labels is impossible for now
    # see https://jira.corezoid.com/browse/COR-6832
    # https://github.com/projectcalico/calico/issues/4017
    - action: Allow
      protocol: TCP
      destination:
        nets:
          - 10.0.0.0/8
          - 172.16.0.0/12
          - 192.168.0.0/16
        ports:
          - 6443
          - 443
